csharp-algorithms/FindAllPositionsOfAString/FindAllPositionsOfAString/Algorithms/SearchUsingIndexOf.cs

```diff
@@ -0,0 +1,25 @@
+﻿using FindAllPositionsOfAString.Algorithms.Interfaces;
+
+namespace FindAllPositionsOfAString.Algorithms;
+
+public class SearchUsingIndexOf : SearchBase, ISearcher
+{
+    public List<int> FindAll(string text, string searchText)
+    {
+        List<int> positions = [];
+
+        var skipSize = ((SkipWholeFoundText) ? searchText.Length : 1);
+
+        var index = text.IndexOf(searchText, StringComparison);
+        while (index != -1)
+        {
+            positions.Add(index);
+            index = text.IndexOf(searchText, index + skipSize, StringComparison);
+        }
+
+        return positions;
+    }
```

Comment:
This will be more performant if we take spans on the arguments and operate over those instead of over the strings. The compiler/runtime can perform additional optimizations in the IndexOf methods when the arguments are spans

Reply:
IndexOf() method operating on ReadOnlySpan<char> does not have  an option to search from N-position

+ If I change the arguments in the interface from string to ReadOnlySpan<char> (which is ugly, for me), then I would have to again convert from ReadOnlySpan<char> to string in RegEx algorithms.

Reply:
1. No need to change the method arguments, just take a span within the method.
2.  You don't need an N-position in the index of, it is a span, so you just slice it:

```
public List<int> FindAll(string text, string searchText)
{
    var spanText = text.AsSpan();
    var spanSearch = searchText.AsSpan();
    List<int> positions = [];

    var skipSize = ((SkipWholeFoundText) ? spanSearch.Length : 1);

    var offset = 0;
    var index = spanText.IndexOf(spanSearch, StringComparison);
    while (index != -1)
    {
        positions.Add(index+offset);
        offset += index + skipSize;
        spanText = spanText[(index + skipSize)..];
        index = spanText.IndexOf(spanSearch, StringComparison);
    }

    return positions;
}
```
The above code can probably be improved and made more readable around the offset computation, etc, but it should give a push in the right direction.

Benchmark using the short string from the pull request:
| Method               | Mean     | Error    | StdDev   | Gen0   | Gen1   | Allocated |
|--------------------- |---------:|---------:|---------:|-------:|-------:|----------:|
| BenchmarkIndexOf     | 68.20 ns | 0.812 ns | 0.720 ns | 0.0459 | 0.0001 |      96 B |
| BenchmarkIndexOfSpan | 62.65 ns | 0.379 ns | 0.296 ns | 0.0459 | 0.0001 |      96 B |